app/src/main/kotlin/com/simplemobiletools/calculator/operation/CurrencyConversionOperation.kt

```diff
@@ -0,0 +1,33 @@
+package com.simplemobiletools.calculator.operation
+
+import com.beust.klaxon.JsonObject
+import com.beust.klaxon.Parser
+
+class CurrencyConversionOperation(var value: Double, var conversion_from: String,
+                                           var conversion_to: String, conversionRatesJsonString: String) {
+
+    var rates = (Parser().parse(conversionRatesJsonString!!.reader()) as JsonObject).obj("rates")!!
+
+    internal fun getRate(): Double {
+        if(conversion_from == conversion_to) {
+            return 1.0
+        } else if(conversion_from == "CAD") {
```

Comment:
Why do specifically hardcode rules for CAD this shouldn't be a thing

Reply:
Our conversion data from the API uses CAD as a base, therefore in-app conversion involving a CAD currency involves less math than conversion between two non-CAD currencies. The alternative would be to use the API data to build conversion ratios between every combination of currencies, which would take more memory and more cycles.

Reply:
Will modify this tomorrow @almiche will be easy 4 temp files for each base currency!